perf/run-perf.sh

```diff
@@ -0,0 +1,221 @@
+#
+
+set -x
+
+#
+# path where all openssl versions are installed.
+# each version should be installed to its prefix.
+# script expects prefix to be something like
+#	/path/to/openssl.installs/openssl-$version
+# 
+# OPENSSL_BINARIES=~/work.openssl/openssl.binaries/
+#
+if [ -z ${OPENSSL_BINARIES+x} ] ; then
+    echo "OPENSSL_BINARIES env. variable not set"
+    exit 1
+fi
+
+for VERSION in master 3.0 1.1.1 ; do
+    OPENSSL_DIR=$OPENSSL_BINARIES/openssl-$VERSION
+    if [ ! -d $OPENSSL_DIR ] ; then
+        echo "OpenSSL $VERSION not found ($OPENSSL_DIR does not exit"
+        exit 1
+    fi
+done
+
+
+#
+# directory where perf tools sources live
+#
+# TOOLS_PATH=~/work.openssl/tools.sashan/perf
+#
+if [ -z ${TOOLS_PATH+x} ] ; then
+    echo "TOOLS_PATH env. variable not set"
+    exit 1
+fi
+
+
+#
+# directory where clone of github repository lives
+# handshake test (et. al.) use certificates from
+# there.
+#
+# OPENSSL_SRC=~/work.openssl/openssl/
+#
+if [ -z ${OPENSSL_SRC+x} ] ; then
+    echo "OPENSSL_SRC env. variable not set"
+    exit 1
+fi
+CERT_DIR=$OPENSSL_SRC/test/certs
+if [ ! -d $CERT_DIR ] ; then
+    echo "$CERT_DIR not found in $OPENSSL_SRC"
+    exit 1
+fi
+
+#
+# directory where to store results
+#
+RESULTS=results
+#
+# Iterations to run each tool
+#
+ITERATIONS=25
+
+function openssl.env {
+    OPENSSL_VERSION=$1;
+    OPENSSL_DIR=$OPENSSL_BINARIES/openssl-$OPENSSL_VERSION
+    if [ -d $OPENSSL_DIR ] ; then
+        #
+        # OpenBSD build (and perhaps other place 64-bit version
+        # to prefix/lib. while for example Debian build installs
+        # openssl under prefix/lib64
+        #
+        if [ -d $OPENSSL_DIR/lib ] ; then
+            LD_LIBRARY_PATH=$OPENSSL_DIR/lib
+            # set them for perf tools testing
+            TARGET_OSSL_LIBRARY_PATH=$OPENSSL_DIR/lib
+            TARGET_OSSL_INCLUDE_PATH=$OPENSSL_DIR/include
+            OPENSSL_HEADERS=$OPENSSL_DIR/include
+            OPENSSL_LIB_PATH=$OPENSSL_DIR/lib
+        elif [ -d $OPENSSL_DIR/lib64 ] ; then
+            LD_LIBRARY_PATH=$OPENSSL_DIR/lib64
+            # set them for perf tools testing
+            TARGET_OSSL_LIBRARY_PATH=$OPENSSL_DIR/lib64
+            TARGET_OSSL_INCLUDE_PATH=$OPENSSL_DIR/include
+            OPENSSL_HEADERS=$OPENSSL_DIR/include
+            OPENSSL_LIB_PATH=$OPENSSL_DIR/lib64
+        else
+            echo "$OPENSSL_DIR does not exist"
+            exit 1
+        fi
+    else
+        echo "$OPENSSL_DIR does not exist"
+        exit 1
+    fi
+}
+
+mkdir -p $RESULTS
+rm -rf $RESULTS/*
+
+#
+# execute all tools we have to convey performance test.
+# each program we run prints single number: microseconds
+# it took to run it.
+#
+for TOOL in evp_fetch randbytes handshake sslnew newrawkey rsasign \
+        x509storeissuer providerdoall rwlocks-rlock rwlocks-wlock \
+	pkeyread-dh-der pkeyread-dhx-der pkeyread-dsa-der pkeyread-ec-der \
+	pkeyread-rsa-der pkeyread-x25519-der pkeyread-dh-pem pkeyread-dhx-pem \
```

Comment:
is the intent here to add scripts or something to represent utilties like pkeyread-dh-der?  I ask because I don't see these utilities added anywhere.  It would actually be reasonable to do as that would let us add scripts that set various options for the tool in different configurations. (handshake -s for example)

Reply:
those utilities were added as symlinks to `pkeyread` tool in PR here: https://github.com/openssl/tools/pull/203
and similarly here for rwlocks: https://github.com/openssl/tools/pull/205